Index: default.py
===================================================================
--- default.py	(revision 43)
+++ default.py	(working copy)
@@ -153,6 +153,9 @@
 
 def gameHighlights():
         thumb = 'http://mlbmc-xbmc.googlecode.com/svn/icons/highlights.png'
+        addGameDir(__language__(30010),dateStr.day[0],26,thumb)
+        addGameDir(__language__(30011),dateStr.day[1],26,thumb)
+        addGameDir(__language__(30012),dateStr.day[3],26,thumb)
         addGameDir(__language__(30036),'8879838',1,thumb)
         addGameDir(__language__(30037),'9781914',1,thumb)
         addGameDir(__language__(30038),'10025018',1,thumb)
@@ -274,7 +277,7 @@
             addPlaylist(name,url,5,'http://mlbmc-xbmc.googlecode.com/svn/icons/tvideo.png')
 
 
-def getRealtimeVideo(url):
+def getRealtimeVideo(url, addYes=True):
         try:
             soup = BeautifulStoneSoup(getRequest(url), convertEntities=BeautifulStoneSoup.XML_ENTITIES)
             videos = soup.findAll('media')
@@ -287,7 +290,8 @@
                 addLink(name,'http://mlb.mlb.com/gen/multimedia/detail/'+url+'.xml',duration,2,thumb)
         except:
             pass
-        addDir(__language__(30017),'http://gdx.mlb.com/components/game/mlb/'+dateStr.day[1]+'/media/highlights.xml',8,icon)
+        if addYes:
+            addDir(__language__(30017),'http://gdx.mlb.com/components/game/mlb/'+dateStr.day[1]+'/media/highlights.xml',8,icon)
 
 
 def getTeamVideo(url):
@@ -403,6 +407,25 @@
             except:
                 continue
 
+def getGameSpecificHighlights(dstr):
+        base = 'http://www.mlb.com/gdcross/components/game/mlb/'
+        thumb = 'http://mlbmc-xbmc.googlecode.com/svn/icons/highlights.png'
+        try:
+       	    data = json.loads(getRequest(base + dstr +'/grid.json'))
+            items = data['data']['games']['game']
+            for i in items:
+                try:
+                    gameId = i['id']
+                    gid = gameId.replace('/','_')
+                    gid = gid.replace('-','_')
+                    glbl = TeamCodes[i['away_team_id']][0] + ' @ ' + TeamCodes[i['home_team_id']][0]
+                    gurl = 'http://gdx.mlb.com/components/game/mlb/' + dstr + '/gid_' + gid + '/media/highlights.xml'
+                    addon_log( "gsh item: " + str(gid) + ', lbl: ' + glbl + ', url:' + gurl )
+                    addDir(glbl, gurl, 27, thumb)
+                except:
+                    continue
+        except:
+            return
 
 def getVideoListXml(url):
         url = 'http://mlb.mlb.com'+url
@@ -1182,4 +1205,10 @@
 if mode==25:
     mlbGame(event, True)
 
-xbmcplugin.endOfDirectory(int(sys.argv[1]))
\ No newline at end of file
+if mode==26:
+    getGameSpecificHighlights(url)
+
+if mode==27:
+    getRealtimeVideo(url, False)
+
+xbmcplugin.endOfDirectory(int(sys.argv[1]))